ngraph/python/src/pyngraph/discrete_type_info.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "pyngraph/discrete_type_info.hpp"
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+
+#include "ngraph/type.hpp"
+
+namespace py = pybind11;
+
+void regclass_pyngraph_DiscreteTypeInfo(py::module m) {
+    py::class_<ngraph::DiscreteTypeInfo, std::shared_ptr<ngraph::DiscreteTypeInfo>> discrete_type_info(
+        m,
+        "DiscreteTypeInfo");
+    discrete_type_info.doc() = "ngraph.impl.DiscreteTypeInfo wraps ngraph::DiscreteTypeInfo";
+    discrete_type_info.def(
+        "__lt__",
+        [](const ngraph::DiscreteTypeInfo& self, const ngraph::DiscreteTypeInfo& other) {
+            return self.version < other.version || (self.version == other.version && strcmp(self.name, other.name) < 0);
+        },
+        py::is_operator());
+    discrete_type_info.def(
+        "__le__",
+        [](const ngraph::DiscreteTypeInfo& self, const ngraph::DiscreteTypeInfo& other) {
+            return self.version < other.version ||
```

Comment:
Looks like you're duplicating code from `DiscreteTypeInfo::operator<=` here. We should be able to reuse these `operator` functions here, rather than recreating their logic again.